src/options.rs

```diff
@@ -136,25 +140,50 @@ impl Options {
     )
   }
 
-  pub(crate) fn bitcoin_rpc_client(&self) -> Result<Client> {
-    let cookie_file = self
-      .cookie_file()
-      .map_err(|err| anyhow!("failed to get cookie file path: {err}"))?;
+  fn derive_var(
+    arg: Option<OsString>,
+    env: Option<OsString>,
+    config: Option<OsString>,
+    default: Option<OsString>,
```

Comment:
The current function signature doesn't allow passing, for example, strings, osstr, cstr, etc. We should make the function signature more generic, so the caller can pass any combination of types that can be converted into an osstr, which will allow fewer conversions at the call site.

I was thinking of suggesting this:

```suggestion
    arg: Option<impl AsRef<OsStr>>,
    env: Option<impl AsRef<OsStr>>,
    config: Option<impl AsRef<OsStr>>,
    default: Option<impl AsRef<OsStr>>,
```

But use of generics forces the compiler to emit a monomerized function for every combination of argument types.

I'm leaning towards this, which should be faster to compile, and which is a bit simpler to read:

```suggestion
    arg: Option<&OsStr>,
    env: Option<&OsStr>,
    config: Option<&OsStr>,
    default: Option<&OsStr>,
```